src/java.security.jgss/share/classes/sun/security/krb5/internal/KRBError.java

```diff
@@ -488,39 +490,34 @@ public byte[] asn1Encode() throws Asn1Exception, IOException {
             return true;
         }
 
-        if (!(obj instanceof KRBError)) {
+        if (!(obj instanceof KRBError other)) {
             return false;
         }
 
-        KRBError other = (KRBError)obj;
         return  pvno == other.pvno &&
                 msgType == other.msgType &&
-                isEqual(cTime, other.cTime) &&
-                isEqual(cuSec, other.cuSec) &&
-                isEqual(sTime, other.sTime) &&
-                isEqual(suSec, other.suSec) &&
                 errorCode == other.errorCode &&
-                isEqual(crealm, other.crealm) &&
-                isEqual(cname, other.cname) &&
-                isEqual(sname, other.sname) &&
-                isEqual(eText, other.eText) &&
-                java.util.Arrays.equals(eData, other.eData) &&
-                isEqual(eCksum, other.eCksum);
-    }
-
-    private static boolean isEqual(Object a, Object b) {
-        return (a == null)?(b == null):(a.equals(b));
+                Objects.equals(cTime, other.cTime) &&
+                Objects.equals(cuSec, other.cuSec) &&
+                Objects.equals(sTime, other.sTime) &&
+                Objects.equals(suSec, other.suSec) &&
+                Objects.equals(crealm, other.crealm) &&
+                Objects.equals(cname, other.cname) &&
+                Objects.equals(sname, other.sname) &&
+                Objects.equals(eText, other.eText) &&
+                Arrays.equals(eData, other.eData) &&
+                Objects.equals(eCksum, other.eCksum);
     }
 
     @Override public int hashCode() {
         int result = 17;
         result = 37 * result + pvno;
         result = 37 * result + msgType;
+        result = 37 * result + errorCode;
         if (cTime != null) result = 37 * result + cTime.hashCode();
         if (cuSec != null) result = 37 * result + cuSec.hashCode();
         if (sTime != null) result = 37 * result + sTime.hashCode();
         if (suSec != null) result = 37 * result + suSec.hashCode();
-        result = 37 * result + errorCode;
         if (crealm != null) result = 37 * result + crealm.hashCode();
         if (cname != null) result = 37 * result + cname.hashCode();
         if (sname != null) result = 37 * result + sname.hashCode();
```

Comment:
Why changing the ordering? I was wondering if you'd change this to Objects.hash(...) with all the fields as arguments.

Reply:
> Why changing the ordering?

The order was changed to be parallel with that of `equals`.

> I was wondering if you'd change this to Objects.hash(...) with all the fields as arguments.

Sure, we could use a convenience method, but because `eData` is an array, it should be `Arrays.deepHashCode(..., eData, ...)` or this construct:

    Objects.hash(..., Arrays.hashCode(eData), ...)

Bottom line: as long as we include all eData elements in the computation, we should be fine. If we fail to do that, we'll violate the equals-hashCode contract.

Reply:
Ok.